libraries/ts/src/margin/marginAccount.ts

```diff
@@ -682,7 +687,7 @@ export class MarginAccount {
   ///
   /// Returns the instruction, and the address of the token account to be
   /// created for the position.
-  async withRegisterPosition(instructions: TransactionInstruction[], tokenMint: Address): Promise<PublicKey> {
+  async withRegisterPosition(instructions: TransactionInstruction[], tokenMint: Address): Promise<TransactionInstruction[]> {
```

Comment:
You don't need to return anything because it's handled in `instructions.push(ix)` at the end of the method
```suggestion
  async withRegisterPosition(instructions: TransactionInstruction[], tokenMint: Address): Promise<void> {
```